### PR TITLE
fix/set saved simplification

### LINKED ISSUE #119: Configuration Option for Reload Prompt
There is a `beforeunload` eventListener which prompts the user to confirm navigation on unsaved changes. 
There's a number of ways to implement ULabel into a project whilst allowing it to persist through several different annotation jobs, and depending on how they set the `saved` state this can break them. 

Disabling this eventListener should be a configuration option, and additionally some reload/save utilities might be useful to streamline how different use cases manage state/save-state whilst doing soft/hard reloads.

### PR DESCRIPTION
**Pull request by joshua-dean**
# Simplification of `set_saved` for submit buttons

## Description
Closes #119. Between this and event listeners being in the `.ulabel` group, implementers of the tool should have full control over the reload prompt. 

I left `set_saved` as optional in the type, as the `SubmitButtons` toolbox item will set the default regardless. 

tl;dr is `set_saved` on submit buttons now defaults to "false". If set to "true", it will call `ULabel.set_saved`, skipping the reload/nav prompt. 

## PR Checklist

- [X] Merged latest main
- [X] Version number in `package.json` has been bumped since last release
- [X] Version numbers match between package `package.json` and `src/version.js`
- [X] Ran `npm install` and `npm run build` AFTER bumping the version number
- [X] Updated documentation if necessary (currently just in `api_spec.md`)
- [X] Added changes to `changelog.md`

## Breaking API Changes
Technically the ability to make a submit button invalidate existing changes and force the reload prompt is gone. 
If anyone was leveraging this functionality, let me know and I'll personally implement a feature to invalidate the save state on submit.

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -106,6 +106,10 @@ export type ULabelSubmitButton = {
     name: string;
     hook: (submit_data: ULabelSubmitData) => void;
     color?: string;
+    /**
+     * Whether or not the button should call `set_saved`
```
Comment (TrevorBurgoyne) [inline index.d.ts:110]:
```suggestion
     * Whether or not the button should call `set_saved(true)`
```

**Review by TrevorBurgoyne — APPROVED** [review]
